rfcs/2021-05-17-7469-load-balancing-aggregators.md

```diff
@@ -0,0 +1,99 @@
+# RFC 7469 - 2021-05-16 - Scaling and load balancing for Vector aggregators
+
+This RFC describes the need for a tooling and environment agnostic load balancing solution to be bundled with Vector aggregator deployments.
+
+* [Scope](#scope)
+* [Motivation](#motivation)
+* [Internal Proposal](#internal-proposal)
+* [Rationale](#rationale)
+* [Prior Art](#prior-art)
+* [Drawbacks](#drawbacks)
+* [Alternatives](#alternatives)
+  * [Do Nothing](#do-nothing)
+  * [Only Client-side Load Balancing](#only-client-side-load-balancing)
+  * [Require a Service Mesh](#require-a-service-mesh)
+* [Outstanding Questions](#outstanding-questions)
+* [Plan Of Attack](#plan-of-attack)
+
+## Scope
+
+Load balancing will be a concern for any `sink` or `source` supported by Vector; some of which can use a general solution (ex: load balancing for our HTTP based `sinks`) and some of which are specific to the component (ex: Kafka `source`/`sink`). Due to the breadth of the topic, this RFC will focus on three specific cases for load balancing while also giving consideration to future adoption for other components.
+
+* Vector agent to Vector aggregator
+* Datadog agent to Vector aggregator
+* Syslog agent to Vector aggregator
```

Comment:
Notably I'm working a `fluent` and `logstash` sources this sprint which it'd be nice to support as an input for aggregators. I don't think it should be any different than supporting the other TCP-based sources though (like `syslog`).

Reply:
Yeah, assuming the proxy supports the protocol we'd just need to wire things up to connect to the right ports